spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/OpenTelemetryAgentExemplarAutoConfiguration.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.springframework.boot.actuate.autoconfigure.tracing;
+
+import io.prometheus.client.exemplars.tracer.otel_agent.OpenTelemetryAgentSpanContextSupplier;
+import org.springframework.boot.actuate.autoconfigure.metrics.export.prometheus.PrometheusMetricsExportAutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * This {@link EnableAutoConfiguration Auto-configuration} runs if the
+ * application has been started with the OpenTelemetry agent.
+ * <p>
+ * Currently, it provides an Exemplar Sampler for Prometheus
+ * based on the agent's tracing, which is why it needs to run
+ * before {@link PrometheusMetricsExportAutoConfiguration}.
+ */
+@AutoConfiguration(before = PrometheusMetricsExportAutoConfiguration.class)
+@ConditionalOnEnabledTracing
+@ConditionalOnClass(name = "io.opentelemetry.javaagent.shaded.io.opentelemetry.api.trace.Span")
```

Comment:
I think this should work the other way and the agent should add this configuration (like it does in other scenarios).
I'm not sure that depending on a shaded and internal class name sounds like a good idea.
Also, as far as I remember, not everything is stable in the agent, I guess the presence of this class might be one of those things.

Reply:
@neiser What do you think?

Reply:
@jonatan-ivanov I agree with your idea to close this PR and see how it could be done within the Java Agent. I simply did not have any time yet to investigate this further (except reacting with :eyes:).